gwvolman/lib/dataone/metadata.py

```diff
@@ -81,6 +82,30 @@ def check_dataone_mimetype(self, mimetype):
             return 'application/octet-stream'
         return mimetype
 
+    def set_related_identifiers(self, manifest, resource_map, eml_pid):
+        """
+        Modifies a resource map to include cito:cites for any cited Tale entities
+
+        :param manifest: The Tale's manifest
+        :param resource_map: The resource map that will be submitted to DataONE
+        :param eml_pid: The pid of the EML document
+        :return: The resource map
+        """
+        try:
+            eml_element = resource_map.getObjectByPid(eml_pid)
+            if eml_pid:
```

Comment:
if the `eml_pid` was wrong or null I think the code would die by now. Don't you want to check if `eml_element` exists instead?

Reply:
Ah that's the returning an rdflib URIRef  object and not the actual string representation. On second thought I should probably be catching an IndexError from the call to [getObjectByPid](https://github.com/DataONEorg/d1_python/blob/master/lib_common/src/d1_common/resource_map.py#L321)

Reply:
Thanks for catching that.